### ISSUE TITLE
[no-inline-styles] distinguish dynamic style bindings from literal ones

### ISSUE THREAD

**Issue by JamesHenry** [angular-eslint/angular-eslint#3186]
`no-inline-styles` exists because styling that could live in a stylesheet should live there (based on user preference). Its `allow*` options are keyed on syntax, so with `allowBindToStyle: true` both of these pass:

```html
<div [style.color]="'red'"></div>
<div [style.width.px]="column.width"></div>
```

The first is a static inline style in binding syntax and should be a class. The second is genuinely dynamic and cannot be a class. The rule's rationale objects to the first and not the second, but the option cannot tell them apart.

Proposal: allow style bindings only when the bound value is not a constant. Constant means a string, number, or boolean literal, or a literal map whose values are all literals. Anything containing a property read, call, or pipe is dynamic. This is decidable from the template AST alone, no type information needed.

Open questions:
- New option (e.g. `allowDynamicStyleBindings`) versus changing what `allowBindToStyle` means. A new option avoids a breaking change.
- Whether `allowNgStyle` should get the same treatment.

Related: #3016, #3112.
